x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/CrossClusterAsyncSearchIT.java

```diff
@@ -1102,33 +1261,53 @@ public void testCancellationViaTimeoutWithAllowPartialResultsSetToFalse() throws
+        // wait for search tasks to complete and be unregistered
         assertBusy(() -> {
-            ListTasksResponse taskResponses = client().admin().cluster().prepareListTasks().setDetailed(true).get();
-            List<TaskInfo> asyncSearchTaskInfos = new ArrayList<>();
-            for (TaskInfo task : taskResponses.getTasks()) {
-                if (task.action().contains("search")) {
-                    if (task.description().contains("async_search{indices[")) {
-                        asyncSearchTaskInfos.add(task);
-                    }
-                }
-            }
-
-            if (asyncSearchTaskInfos.size() > 0) {
-                // if still present, and it is cancelled, then we can proceed with the test
-                assertTrue(asyncSearchTaskInfos.get(0).cancelled());
-            }
-            // if not present, then it has been unregistered and the async search should no longer be running, so can proceed
-        }, 30, TimeUnit.SECONDS);
+            ListTasksResponse listTasksResponse = client(LOCAL_CLUSTER).admin()
+                .cluster()
+                .prepareListTasks()
+                .setActions(SearchAction.INSTANCE.name())
+                .get();
+            List<TaskInfo> tasks = listTasksResponse.getTasks();
+            assertThat(tasks.size(), equalTo(0));
 
-        assertBusy(() -> { assertFalse(getAsyncStatus(response.getId()).isRunning()); });
+            ListTasksResponse remoteTasksResponse = client(REMOTE_CLUSTER).admin()
+                .cluster()
+                .prepareListTasks()
+                .setActions(SearchAction.INSTANCE.name())
+                .get();
+            List<TaskInfo> remoteTasks = remoteTasksResponse.getTasks();
+            assertThat(remoteTasks.size(), equalTo(0));
+        });
 
         AsyncStatusResponse statusResponse = getAsyncStatus(response.getId());
         assertFalse(statusResponse.isRunning());
         assertEquals(0, statusResponse.getSuccessfulShards());
         assertEquals(0, statusResponse.getSkippedShards());
         assertThat(statusResponse.getFailedShards(), greaterThanOrEqualTo(1));
 
+        waitForSearchTasksToFinish();
+    }
+
+    private void waitForSearchTasksToFinish() throws Exception {
+        assertBusy(() -> {
+            ListTasksResponse listTasksResponse = client(LOCAL_CLUSTER).admin()
+                .cluster()
+                .prepareListTasks()
+                .setActions(SearchAction.INSTANCE.name())
+                .get();
+            List<TaskInfo> tasks = listTasksResponse.getTasks();
+            assertThat(tasks.size(), equalTo(0));
+
+            ListTasksResponse remoteTasksResponse = client(REMOTE_CLUSTER).admin()
+                .cluster()
+                .prepareListTasks()
+                .setActions(SearchAction.INSTANCE.name())
+                .get();
+            List<TaskInfo> remoteTasks = remoteTasksResponse.getTasks();
+            assertThat(remoteTasks.size(), equalTo(0));
```

Comment:
NIT: `assertThat(remoteTasks, empty());` for a nicer failure message in case we ever need it :)